myApp/app/context/FeedbackContext.js

```diff
@@ -0,0 +1,48 @@
+import React, { createContext, useContext, useState, useCallback } from 'react';
+
+const FeedbackContext = createContext();
+
+export function FeedbackProvider({ children }) {
+  const [vibrationEnabled, setVibrationEnabled] = useState(true);
+  const [soundEnabled, setSoundEnabled] = useState(true);
+  const [speechEnabled, setSpeechEnabled] = useState(true);
+
+  const toggleVibration = useCallback(() => {
+    setVibrationEnabled(prev => {
+      const newState = !prev;
+      setVibrationEnabled(newState);
+      return newState;
+    });
+  }, []);
+
+  const toggleSound = useCallback(() => {
+    setSoundEnabled(prev => {
+      const newState = !prev;
+      setSoundEnabled(newState);
+      return newState;
+    });
+  }, []);
+
+  const toggleSpeech = useCallback(() => {
+    setSpeechEnabled(prev => {
+      const newState = !prev;
+      setSpeechEnabled(newState);
+      return newState;
+    });
+  }, []);
```

Comment:
```suggestion
const toggleVibration = useCallback(() => {
  setVibrationEnabled(prev => !prev);
}, []);

const toggleSound = useCallback(() => {
  setSoundEnabled(prev => !prev);
}, []);

const toggleSpeech = useCallback(() => {
  setSpeechEnabled(prev => !prev);
}, []);
```
Avoid Redundant State Updates, you can simpilfy by directly returning the new state.

Reply:
Applied the changes!